rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/types.hpp

```diff
@@ -15,6 +15,15 @@
 #ifndef RMW_OPENDDS_SHARED_CPP__TYPES_HPP_
 #define RMW_OPENDDS_SHARED_CPP__TYPES_HPP_
 
+#include "topic_cache.hpp"
+#include "rmw_opendds_shared_cpp/RmwAllocateFree.hpp"
+#include "rmw_opendds_shared_cpp/opendds_include.hpp"
+#include "rmw_opendds_shared_cpp/visibility_control.h"
```

Comment:
```suggestion
#include "RmwAllocateFree.hpp"
#include "opendds_include.hpp"
#include "visibility_control.h"
```

Reply:
For the following reasons, I propose not to make these changes in this PR:

1) Currently the rmw repos use double quotes to include most files except standard library files. This definitely differs from our OpenDDS standard regarding double quotes and angle brackets. The order of includes is even worse. Sometimes when I made substantial changes in some files, I also changed the include quotes and order, but not systematically change all of them yet.

2) The rmw src files are organized in such directories:
rmw_opendds_cpp/include/rmw_opendds_cpp/identifier.hpp
rmw_opendds_cpp/src/identifier.cpp

rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/init.hpp
rmw_opendds_shared_cpp/src/init.cpp

For a cpp file, the same name header file should be first included (in double quotes if in the same directory). However, in rmw_opendds_shared_cpp/src/init.cpp, for example, we have #include "rmw_opendds_shared_cpp/init.hpp". Since they are not in the same directories, should we go ahead replace all of such includes with angle brackets? 

It is not at all unlikely that different rmw implementations have the same filenames, such as identifier.hpp or init.hpp. Do we feel safe to drop the directories ("rmw_opendds_cpp/" or "rmw_opendds_shared_cpp/") from the includes now when we are right in the middle of trying to figure out some problems including why we did get our own implementation_identifier? At least the directories have _opendds_ in it. It's best that we focus on the intended changes here and hold the other things later.

So I'll create a jira card for investigating our rmw includes. (Maybe rmw already has a linter for it.)

Reply:
> For a cpp file, the same name header file should be first included (in double quotes if in the same directory). However, in rmw_opendds_shared_cpp/src/init.cpp, for example, we have #include "rmw_opendds_shared_cpp/init.hpp". Since they are not in the same directories, should we go ahead replace all of such includes with angle brackets?

I think using double quotes include in a cpp in the same repo file is ok since the included file like "rmw_opendds_shared_cpp/init.hpp" is in a relative path to the source file. This is similar to the include rules of OpenDDS.

> It is not at all unlikely that different rmw implementations have the same filenames, such as identifier.hpp or init.hpp. Do we feel safe to drop the directories ("rmw_opendds_cpp/" or "rmw_opendds_shared_cpp/") from the includes now when we are right in the middle of trying to figure out some problems including why we did not get our own implementation_identifier? At least the directories have opendds in it. It's best that we focus on the intended changes here and hold the other things later.

These 3 header files are in the same directory as types.hpp, so I think it's safe to drop the directory path. Line 18 of types.hpp already dropped the directory path for including `topic_cache.hpp`. By the way, I can't find RmwAllocateFree.hpp in include/rmw_opendds_shared_cpp, and looks like it's not a newly created file either?

Reply:
I created a jira card for solving the include issue consistently across the board. Currently there are quite some problems in the system tests we haven't figured out yet. (We didn't run these system tests last year, so these are new found problems.) Many system tests build and run mix-matched implementations (e.g. opendds publisher and fastrtps subscriber, or cpp and py). We now have the statistics of the failed tests, and the current effort is to step by step fix the problems. If we change too many things now without clear expectations toward fixing the problems (or just because we know something can be better here or there), we may lose the consistency and comparisons with the known statistics. So unless we can now link an include statement to a system test error, I still prefer to solve the include issue consistently in another PR. The RmwAllocateFree was merged in PR#53.

Reply:
As a matter of fact, in my previous modifications, I changed a lot of the include statements according to our OpenDDS standard. When I realized the system tests actually try to build and run mixed implementations, I put those directories back in the include statements just to avoid the risk of introducing problems without knowing. (The implementation identifier mismatch is on my mind all the time.) I want to solve the include problem once for all when we have a good state to compare to.

Reply:
This is fine. I guess the includes can be cleaned up later in a separate PR.